### PR TITLE
Unify trove4j version in build

### PR DESCRIPTION
**Pull request by blindpirate**
Also to avoid the verification failure

### CODE REVIEW

**Comment by blindpirate** [comment]
@bot-gradle test RFN

File: subprojects/internal-android-performance-testing/build.gradle.kts [hunk 1]
```diff
@@ -10,7 +10,11 @@ repositories {
 dependencies {
     implementation(project(":base-services"))
     implementation(project(":tooling-api"))
-    api("com.android.tools.build:gradle:3.0.0")
+    api("com.android.tools.build:gradle:3.0.0") {
+        // unify the trove4j version we're using
+        exclude(group = libs.trove4j.substringBefore(":"), module = libs.trove4j.substringAfter(":"))
```
Comment (wolfs) [inline subprojects/internal-android-performance-testing/build.gradle.kts:15]: Please use a metadata rule and not an exclude for this. The dependency of this artifact should be changed to "our" trove4j library. We should then also remove the other trove4j library from the dependency verification file.

**Comment by blindpirate** [comment]
@bot-gradle test this

File: build-logic/dependency-modules/src/main/kotlin/gradlebuild.dependency-modules.gradle.kts [hunk 2]
```diff
@@ -247,3 +249,17 @@ abstract class ReplaceCglibNodepWithCglibRule : ComponentMetadataRule {
...
 }
+
+// https://youtrack.jetbrains.com/issue/IDEA-261387
+abstract class UnifyTrove4jVersionRule : ComponentMetadataRule {
+    override fun execute(context: ComponentMetadataContext) {
+        context.details.allVariants {
+            withDependencies {
+                if (any { it.name == "trove4j" }) {
+                    removeAll { it.name == "trove4j" }
+                    add("org.jetbrains.intellij.deps:trove4j:trove4j:1.0.20181211")
```
Comment (wolfs) [inline build-logic/dependency-modules/src/main/kotlin/gradlebuild.dependency-modules.gradle.kts:260]: We manage the version via our platform. Why do we need to specify the version here again?

File: build-logic/dependency-modules/src/main/kotlin/gradlebuild/modules/extension/ExternalModulesExtension.kt [hunk 3]
```diff
@@ -251,6 +252,7 @@ abstract class ExternalModulesExtension {
         testng to License.Apache2,
         tomlj to License.Apache2,
         trove4j to License.LGPL21,
+        agp to License.Apache2,
```
Comment (wolfs) [inline build-logic/dependency-modules/src/main/kotlin/gradlebuild/modules/extension/ExternalModulesExtension.kt:255]: Do we need this? I thought we only need this for things we bundle in our distribution.

**Comment by blindpirate** [comment]
@bot-gradle test this

File: subprojects/distributions-dependencies/build.gradle.kts [hunk 4]
```diff
@@ -125,6 +125,8 @@ dependencies {
         api(libs.testng)                { version { strictly("6.3.1"); because("later versions break test cross-version test filtering") }}
         api(libs.tomlj)                 { version { strictly(tomljVersion) }}
         api(libs.trove4j)               { version { strictly("1.0.20181211") }}
+        // TODO upgrade this AGP version to recent version
```
Comment (wolfs) [inline subprojects/distributions-dependencies/build.gradle.kts:128]: Actually, we should drop our Android studio sync and use the real thing with the new features in Gradle profiler. Though that may be hard.

File: subprojects/internal-android-performance-testing/build.gradle.kts [hunk 5]
```diff
@@ -1,6 +1,7 @@
 plugins {
     id("gradlebuild.internal.java")
     application
+    id("gradlebuild.dependency-modules")
```
Comment (wolfs) [inline subprojects/internal-android-performance-testing/build.gradle.kts:4]: I am surprised you need to apply this plugin here explicitly. Doesn't the `gradlebuild.internal.java` plugin apply this?

Comment (blindpirate) [reply subprojects/internal-android-performance-testing/build.gradle.kts]: This is a leftover in the development. Removed.

**Review by wolfs — APPROVED** [review]
LGTM!

**Comment by blindpirate** [comment]
@bot-gradle test this

**Pull request merged**